aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -254,6 +271,23 @@ module aptos_token_objects::token {
         }
     }
 
+    /// This method allows minting to happen in parallel, making it efficient.
+    public fun index_snapshot<T: key>(token: &Object<T>): AggregatorSnapshot<u64> acquires Token, TokenAppendix1 {
+        let token_address = object::object_address(token);
+        if (exists<TokenAppendix1>(token_address)) {
+            aggregator_v2::copy_snapshot(&borrow_global<TokenAppendix1>(token_address).index)
+        } else {
+            aggregator_v2::create_snapshot(borrow(token).deprecated_index)
```

Comment:
is this efficient? asking if below, we should just perform the check again.

Reply:
The original motivation to introduce aggregator snapshots is to replace `read(agg)` with `snapshot(agg)`, and materialize the snapshot at a later point in time. How are we achieving that with this `create_snapshot` method? Here, we are creating a snapshot from an index (u64) instead of an aggregator. How is a snapshot helping us here?